### PR TITLE
restructure of rpacket_tracker

### PR DESCRIPTION
**Pull request by Sumit112192**
### :pencil: Description

**Type:**  :roller_coaster: `infrastructure` and `restructure`

This PR restructures the rpacket_tracker code to initialize only when the track_rpacket flag is set. Apart from that, I will also add tests for the rpacket_tracker and change how it registers an interaction.

### CODE REVIEW

**Comment by Sumit112192** [comment]
@wkerzendorf @andrewfullard Please review.

**Comment by Sumit112192** [comment]
Profiling:
Master Branch:
image
FixRPacketTracker Branch:
image
Memory usage has gone done.

File: tardis/conftest.py [hunk 1]
```diff
@@ -243,3 +252,32 @@ def simulation_verysimple_vpacket_tracking(config_verysimple, atomic_dataset):
...
+    Parameters
+    ----------
+    config_verysimple_rpacket_tracking : tardis.io.configuration.config_reader.Configuration
+    atomic_dataset : AtomData
+
+    Returns
+    -------
+    simulation object with track_rpacket enabled
+    """
+    config_rpacket_tracking.montecarlo.iterations = 3
```
Comment (andrewfullard) [inline tardis/conftest.py:271]: Do we need 3 iterations for the test to be useful?

**Review by andrewfullard — APPROVED** [review]

File: tardis/conftest.py (hunk 1 above)
Comment (Sumit112192) [reply tardis/conftest.py:271]: I don't think so. But it shouldn't take too much time.

**Comment by Sumit112192** [comment]
Completet as PR #2716 is merged

**Pull request closed without merging**